src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -516,44 +518,95 @@ public void Initialize(ITelemetry telemetry)
             }
 
             telemetry.Context.Initialize(this.Context, instrumentationKey);
-            foreach (ITelemetryInitializer initializer in this.configuration.TelemetryInitializers)
+
+            ISupportAdvancedSampling telemetryWithSampling = telemetry as ISupportAdvancedSampling;
+
+            // Telemetry can be already sampled out if that decision was made before calling Track()
+            bool sampledOut = telemetryWithSampling?.IsProactivelySampledOut ?? false;
+            bool shouldTryHeadSampling = this.proactiveSamplingEnabled && (telemetryWithSampling?.SupportsProactiveSampling ?? false);
+
+            if (!sampledOut)
             {
-                try
+                if (shouldTryHeadSampling)
                 {
-                    initializer.Initialize(telemetry);
+                    // If telemetry was not sampled out, we need to check its sampling score after we learn its Operation Id                    
+                    bool shouldCheckSamplingScore = true;
+
+                    for (int index = 0; index < this.configuration.TelemetryInitializers.Count; index++)
+                    {
+                        if (shouldCheckSamplingScore && !string.IsNullOrEmpty(telemetry.Context.Operation.Id))
+                        {
+                            if (SamplingScoreGenerator.GetSamplingScore(telemetry.Context.Operation.Id) >= this.configuration.GetLastObservedSamplingPercentage(telemetryWithSampling.ItemTypeFlag))
+                            {
+                                telemetryWithSampling.IsProactivelySampledOut = true;
```

Comment:
please log that we are setting this item to be pro-act-sampled. Would be good to mention the 'index' in the logs as well, so we can track which initializers from the list will be skipped.

Reply:
Its possible to use telemetryClient, with no built-in telemetry processors. so its best to log who is setting IsProactivelySampledout = true. Since its possible that other auto-collector could also be setting this to true.